src/agent/src/initdata.rs

```diff
@@ -0,0 +1,239 @@
+//! # Initdata Module
+//!
+//! This module will do the following things if [`INITDATA_DEV`] exists.
+//! 1. Parse the initdata block device and extract the config files to [`INITDATA_PATH`].
+//! 2. Store the initdata hash in [`INITDATA`].
+
+// Copyright (c) 2024 Alibaba Cloud
+//
+// SPDX-License-Identifier: Apache-2.0
+//
+
+use std::{
+    io::{BufReader, Read},
+    os::unix::fs::FileTypeExt,
+    path::Path,
+};
+
+use anyhow::{bail, Context, Result};
+use backhand::{FilesystemReader, InnerNode};
+use const_format::concatcp;
+use serde::Deserialize;
+use sha2::{Digest, Sha256, Sha384, Sha512};
+use slog::Logger;
+use tokio::{io::AsyncReadExt, sync::OnceCell};
+
+/// This is the target directory to store the extracted initdata.
+pub const INITDATA_PATH: &str = "/run/confidential-containers/initdata";
+
+/// The path of AA's config file
+pub const AA_CONFIG_PATH: &str = concatcp!(INITDATA_PATH, "/aa.toml");
+
+/// The path of CDH's config file
+pub const CDH_CONFIG_PATH: &str = concatcp!(INITDATA_PATH, "/cdh.toml");
+
+/// Initdata file path inside squashfs image
+pub const INITDATA_FILE_IN_SQUASHFS: &str = "/.kata.initdata.toml";
+
+/// Now only initdata `0.1.0` is defined.
+const INITDATA_VERSION: &str = "0.1.0";
+
+/// If initdata is given and parsed successfully, this static
+/// variable will be set with the base64 encoded digest of the
+/// initdata.
+pub static INITDATA: OnceCell<Vec<u8>> = OnceCell::const_new();
```

Comment:
i'd call it `INITDATA_DIGEST_B64` to avoid ambiguity.

Reply:
Oh, the comments are old. Let me fix this, it should be INTIDATA